Начал делать функционал панели с файлами

agafoner opened the pull request.

agafoner committed:
Проблемы с cors

agafoner committed:
1 Вынес интерфейс файловой системы в отдельный файл в api
2 Импортировал интерфейсы в store в client

agafoner committed:
1 Сделал неотформатированный вывод списка файлов/папок.
2 Сделал выбор диска.

StepanovAleksey commented on api/package.json:
```diff
@@ -24,6 +24,7 @@
     "@nestjs/common": "^9.0.0",
     "@nestjs/core": "^9.0.0",
     "@nestjs/platform-express": "^9.0.0",
+    "cors": "^2.8.5",
```
?

StepanovAleksey commented on api/src/services/app.fileSystemService.ts:
```diff
@@ -1,27 +1,14 @@
-    const dir = path.win32.normalize(targetPath);
+    const dir = path.win32.normalize(targetPath || 'C:/');
     let files = [];
     let files_ = [];
```
зачем не нужная переменная если можно сразу возвращать?

StepanovAleksey commented on api/src/services/app.fileSystemService.ts:
```diff
@@ -45,6 +32,7 @@ export class AppFileSystemService {
         return <FileModel | FolderModel> {name: name, lastDateChange:date, size: size, type: "file"}
```
Если у тебя есть проверка то ты уже знаешь тип, зачем указывать оба? в данном случае у тебя `FileModel `

StepanovAleksey replied:
Лучше объявить классы и использовать конструкторы

StepanovAleksey commented on client/src/components/panel/ElList.vue:
```diff
@@ -1,9 +1,9 @@
 <div class="el-List">
-  <FileElement v-for="file in files" :fileData="file"></FileElement>
+
+  <FileElement v-for="row in $props.dirList" v-if="$props.dirList!==null" :row="row"></FileElement>
```
можно просто `$props.dirList` или `!!$props.dirList`

StepanovAleksey commented on client/src/components/panel/PanelBar.vue:
```diff
@@ -1,28 +1,47 @@
-<div class="panelBar">
-  <select>
-    <option v-for="partition in $store.state.diskPart">{{partition}}</option>
+<div class="panelBar inline">
```
`panelBar` придерживайся одного стиля kebabe-case

StepanovAleksey commented on client/src/components/panel/PanelBar.vue:
```diff
@@ -1,28 +1,47 @@
+  props:[ 'panelId' ],
+  data() {
+    return {
+      selectedDisk: 'C:'
```
откуда знаешь значение диска? А если на машине не будет диска C?

agafoner replied:
Не могу сообразить, как реализовать красиво - оставил коммент в коде

StepanovAleksey replied:
ответил в другом комментарии

StepanovAleksey reviewed (commented):
Поправь замечания

agafoner committed:
Сделал нормализацию слешей в пути к папке

agafoner committed:
1 Поправил часть замечаний, остались 2 вопроса:
- по отображению текущего диска в select при открытии/обновлении приложения
- не сделал пока класс-конструктор для моделей папок и файлов, нужно уточнение

agafoner committed:
1 Добавил функционал возвращение в вышестоящую папку

StepanovAleksey commented on api/src/services/app.fileSystemService.ts:
```diff
@@ -39,14 +25,13 @@ export class AppFileSystemService {
       if (stat.isDirectory()) {
         name = name + path.sep
-        return <FileModel | FolderModel> {name: name, lastDateChange: date, type: "folder"}
+        return <FolderModel> {name: name, lastDateChange: date, type: "folder"}
```
а где классы?

StepanovAleksey commented on client/src/components/Panel.vue:
```diff
@@ -16,7 +17,10 @@ import {defineComponent} from "vue";
-  props:[ 'panelId' ]
+  props:[ 'panelId' ],
+  mounted() {
+       this.$store.state.fetchDirList('C:/',this.panelId)
```
я бы на твоём месте сделал по другому. При получении данных под дискам выставлял бы во всех панелях первый диск их списка и уже после этого слал запрос по этому диску

StepanovAleksey commented on client/src/components/panel/ElList.vue:
```diff
@@ -12,14 +14,40 @@ import {defineComponent} from "vue";
+
+  methods: {
+    selectDir: function (name: string, type: string) {
+      if (type!='folder') return
```
!==

StepanovAleksey commented on client/src/components/panel/ElList.vue:
```diff
@@ -12,14 +14,40 @@ import {defineComponent} from "vue";
+        console.log('upperDir')
+        const str=this.$store.state.currentDir[this.$props.panelId]
+        if (!str) return false
+        let arr=str.replaceAll('\\','/').split('/')
```
ты хранишь историю в ввиду одной строки. но на мой взгляд тебе было бы проще если бы ты хранил её в ввиде массива и тогда не пришлось ничего парсить

StepanovAleksey commented on client/src/components/panel/FileElement.vue:
```diff
@@ -1,8 +1,17 @@
-  {{fileData.name}}
-  {{fileData.size}}
-  {{fileData.date}}
+<div class="element inLine" @dblclick="$emit('chDir',row.name, row.type)">
```
объяви событие в компоненте

StepanovAleksey commented on client/src/components/panel/PanelBar.vue:
```diff
@@ -1,28 +1,57 @@
-    <option v-for="partition in $store.state.diskPart">{{partition}}</option>
+<template v-if="!!this.$store.state.currentDir[panelId]">
+<div class="panel-bar inline">
+  <select v-model="selectedDisk" @change="selectDisk(selectedDisk)" > <!-- Не могу сообразить, как сделать, чтобы при монтировании в select'e нужный диск. Пробовал вотчером и через computed - получается лабуда. -->
```
храни выбранные диск в сторе, и через стор меняй

StepanovAleksey commented on client/src/store.ts:
```diff
@@ -1,19 +1,48 @@
 }
+const api=axios.create({
+    baseURL: 'http://127.0.0.1:5000/api/',
+    headers:  { 'Content-Type': 'application/json', "Access-Control-Allow-Methods": "GET","Access-Control-Allow-Origin":"*"  }
```
"Access-Control-Allow-Origin Access-Control-Allow-Methods по моему теперь не обязательны

StepanovAleksey commented on client/src/store.ts:
```diff
@@ -1,19 +1,48 @@
+         api
+             .get(state.endPoint.dirList+'?path='+dir)
+             .then(response => {state.dirList[panel]=response.data; }).catch(e=>(console.log('Error fetchDirList'+e)))
+         state.currentDir[panel]=dir;
```
тут нельзя через this обращаться?

StepanovAleksey reviewed (commented):
добавил  новые замечания

agafoner committed:
Не получилось: PanelBar - стал обращаться в стор, но всё-равно при запуске диск не выбран

StepanovAleksey committed:
немного поигрался со store.ts

agafoner committed:
1 Сделал переход в директорию по двойному клику
2 Сделал строчку возврата в директорию выше, если это возможно.

agafoner committed:
Добавил возможность выделения файлов по клику
